compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -2146,13 +2146,15 @@ object Types {
   }
 
   case class LazyRef(refFn: () => Type) extends UncachedProxyType with ValueType {
+    private var myRefFn: () => Type = refFn
```

Comment:
I am not sure it would work in general, though. Seems generally dicy to evaluate in a different context. We'd have to check every site where we create a LazyRef whether it is permissible.